microprofile-health/src/main/java/org/quickstart/microprofile/health/DatabaseConnectionHealthCheck.java

```diff
@@ -0,0 +1,42 @@
+package org.quickstart.microprofile.health;
+
+import org.eclipse.microprofile.config.inject.ConfigProperty;
+import org.eclipse.microprofile.health.HealthCheck;
+import org.eclipse.microprofile.health.HealthCheckResponse;
+import org.eclipse.microprofile.health.HealthCheckResponseBuilder;
+import org.eclipse.microprofile.health.Readiness;
+
+import javax.enterprise.context.ApplicationScoped;
+import javax.inject.Inject;
+
+@Readiness
+@ApplicationScoped
+public class DatabaseConnectionHealthCheck implements HealthCheck {
```

Comment:
I'm not sure that a database connection health check is correct for this quickstart.
To me, it is an anti pattern. Health checks should check the readiness/healthiness of the deployment itself without crossing service boundaries (and the database is another service in that case).

If the application is not considered healthy with that regard, it will be restarted but this will not fix the database connection issue. In production, this would likely cause a cascade of failure/restart from the application without any benefit.

Could the quickstart instead check (and simulate failure) on in-vm resources (thread pools, memory consumption, cpu activity, etc.)?

Reply:
This is a readiness check which in my opinion is directly targeted for this kind of scenario. Readiness check won't restart the container, it just prohibits the traffic. This is the main difference between liveness and readiness. From https://kubernetes.io/docs/tasks/configure-pod-container/configure-liveness-readiness-startup-probes/#define-readiness-probes:

Sometimes, applications are temporarily unable to serve traffic. For example, an application might need to load large data or configuration files during startup, or depend on external services after startup. In such cases, you don’t want to kill the application, but you don’t want to send it requests either.

Reply:
ah, you're right. I did not pay attention that it was a readiness probe and not a liveness probe.

looks good to me